lib/floe/workflow/choice_rule/data.rb

```diff
@@ -22,55 +22,120 @@ def initialize(_workflow, _name, payload)
           parse_compare_key
         end
 
+        # Evaluate whether this rule is true for the given context and input (runtime)
+        # @param context [Context] The workflow execution context
+        # @param input [Hash] The current state input
+        # @return [Boolean] true if the rule evaluate to true
         def true?(context, input)
-          return presence_check(context, input) if compare_key == "IsPresent"
-
-          lhs = variable_value(context, input)
+          # right hand side of comparison or predicate if this a type check
           rhs = compare_value(context, input)
+          # left hand side of comparison or value of the variable if this is a type check
+          lhs = variable_value(context, input)
+          # Example:
+          #
+          # {"Variable": "$.foo", "IsNumeric": true}
+          #   value     = lhs = input["$.foo"]
+          #   predicate = rhs = true
+          #               true meaning we expect $.foo to be numeric.
+          # => is_numeric?(value, predicate)
+          #
+          # {"Variable": "$.foo", "GreaterThanString": "aaa"}
+          #   lhs = input["$.foo"]
+          #   rhs = "aaa"
+          # => op_greaterthan?(lhs, rhs)
+          #
+          # {"Variable": "$.foo", "GreaterThanNumericPath": "$.bar"}
+          #   lhs = input["$.foo"]
+          #   rhs = input["$.bar"]
+          # => op_greaterthan?(lhs, rhs)
+          #
```

Comment:
I think you can simplify this comment to 


```suggestion
          # Payload pattern is: {"Variable": $lhs, $operator: $rhs}
          # Example:
          #
          #   {"Variable": "$.foo", "IsNumeric": true}
          #     lhs = input["$.foo"]
          #     rhs = true
          #     is_numeric?(lhs, rhs)
          #
          #   {"Variable": "$.foo", "GreaterThanString": "aaa"}
          #     lhs = input["$.foo"]
          #     rhs = "aaa"
          #     op_greaterthan?(lhs, rhs)
          #
          #   {"Variable": "$.foo", "GreaterThanNumericPath": "$.bar"}
          #     lhs = input["$.foo"]
          #     rhs = input["$.bar"]
          #     op_greaterthan?(lhs, rhs)
          #
          # NOTE: IsPresent works a little differently as the lhs might raise a PathError. See the exaception handler below.
          rhs = compare_value(context, input)
          lhs = variable_value(context, input)
```